packages/seacas/libraries/ioss/src/text_mesh/Iotm_TextMeshUtils.h

```diff
@@ -14,593 +14,1754 @@
+  }
+
+  const std::vector<double> &operator[](const EntityId nodeId) const
+  {
+    auto it(m_nodalCoords.find(nodeId));
+
+    if (it == m_nodalCoords.end()) {
+      std::ostringstream errmsg;
+      errmsg << "Could not find node id " << nodeId;
+      m_errorHandler(errmsg);
+    }
 
-        for (auto iter : m_parts) {
-          names.push_back(iter.second);
-        }
+    return it->second;
+  }
 
-        return names;
-      }
+  void set_coordinate_data(
+      const unsigned spatialDim, const std::set<EntityId> &nodeIds, const std::vector<double> &coordinates)
+  {
+    if (!coordinates.empty()) {
+      validate_num_coordinates(spatialDim, nodeIds, coordinates);
+      fill_coordinate_map(spatialDim, nodeIds, coordinates);
+      m_hasCoordinateData = true;
+    }
+  }
 
-      void set_error_handler(ErrorHandler errorHandler) { m_errorHandler = errorHandler; }
+  void set_error_handler(ErrorHandler errorHandler) { m_errorHandler = errorHandler; }
 
-    private:
-      void handle_block_part(const std::string &name)
-      {
-        const std::string blockPrefix  = "BLOCK_";
-        const unsigned    prefixLength = blockPrefix.length();
+  bool has_coordinate_data() const { return m_hasCoordinateData; }
 
-        if (name.length() < prefixLength + 1)
-          return;
+ private:
+  void validate_num_coordinates(
+      const unsigned spatialDim, const std::set<EntityId> &nodeIds, const std::vector<double> &coordinates)
+  {
+    if (coordinates.size() != nodeIds.size() * spatialDim) {
+      std::ostringstream errmsg;
+      errmsg << "Number of coordinates: " << coordinates.size() << ", Number of nodes: " << nodeIds.size()
+             << ", Spatial dimension: " << spatialDim;
+      m_errorHandler(errmsg);
+    }
+  }
+
+  void fill_coordinate_map(
+      const unsigned spatialDim, const std::set<EntityId> &nodeIds, const std::vector<double> &coordinates)
+  {
+    std::vector<double>::const_iterator coordIter = coordinates.begin();
+    for (const EntityId &nodeId : nodeIds) {
+      m_nodalCoords[nodeId] = std::vector<double>(coordIter, coordIter + spatialDim);
+      coordIter += spatialDim;
+    }
+  }
+
+  bool m_hasCoordinateData{false};
+  std::unordered_map<EntityId, std::vector<double>> m_nodalCoords;
+  ErrorHandler m_errorHandler;
+};
+
+template <typename EntityId, typename Topology>
+struct ElementData {
+  int proc;
+  EntityId identifier;
+  Topology topology;
+  std::vector<EntityId> nodeIds;
+  std::string partName = "";
+
+  operator EntityId() const { return identifier; }
+};
+
+struct SideBlockInfo {
+  std::string name;
+  std::string sideTopology;
+  std::string elementTopology;
+  std::string touchingBlock;
+  std::vector<size_t> sideIndex;
+  unsigned numNodesPerSide;
+};
+
+enum SplitType { SPLIT_BY_TOPOLOGIES = 1, SPLIT_BY_ELEMENT_BLOCK = 2, SPLIT_BY_DONT_SPLIT = 3, SPLIT_INVALID = 4 };
+
+inline std::ostream &operator<<(std::ostream &out, const SplitType &t)
```

Comment:
If this is needed, should be added to Ioss_SurfaceSplit.h for consistency and robustness.  Two definitions in two places will probably lead to then getting out of sync.

Reply:
Please see above. The options are definitely related to Ioss but don't necessarily have to be the same